docs/source/sci_api_tutorial/preparation_protonate_stru.rst

```diff
@@ -0,0 +1,185 @@
+
+Arguments
+==============================================
+
+``stru``
+    The input ``Structure`` instance (no matter it's a protein, polypeptite, or ligand).
+
+``ph``
+    The pH value for determining the protonation state.
+    (Float, optional, default ``7.0``)
+
+    .. dropdown:: :fa:`eye,mr-1` Click to learn more about ``ph``
+
+        The choice of pH value depends on the physiological environment of the protein you are simulating.
+
+        For example, if you are simulating a protein in human blood, then the pH value is best set in the range [7.35, 7.45].
+
+``protonate_ligand``
+    If also protonate ligand.
+    
+    (Boolean, optional, default ``True``)
+
+``engine``
+    Engine for determining the pKa and adding hydrogens to the protein peptide part.
+
+    (String, optional, default ``pdb2pqr``) 
+    
+    .. dropdown:: :fa:`eye,mr-1` Click to learn more about ``engine``
+
+        The ``engine`` option determines which third-party tool you use to add hydrogen atoms to your protein (polypeptide).
+
+        Currently only ``pdb2pqr`` is available. More ``engine`` options will be available in future versions.
+
+``ligand_engine``
+    Engine for adding hydrogens to ligands (current available values: ``pybel``)
+
+    (String, optional, default ``pybel``)
+    
+    .. dropdown:: :fa:`eye,mr-1` Click to learn more about ``engine``
+
+        The ``ligand_engine`` option determines which third-party tool you use to add hydrogen atoms to your ligand.
+
+        Currently only ``pybel`` is available. More ``ligand_engine`` options will be available in future versions.
+
+``**kwarg``
+    Setting(s)/Option(s) related to specific engine.
+
+    (optional, default ``None``)
+
+Examples
+==============================================
+
+Prepare the Input: Load Structure
+----------------------------------------------
+
+.. panels::
+
+    :column: col-lg-12 col-md-12 col-sm-12 col-xs-12 p-2 text-left
+
+    In order to make use of the API, we should have structure loaded.
+
+    .. code:: python    
+
+        import enzy_htp.structure as struct
+                                    
+        sp = struct.PDBParser()
+
+        pdb_filepath = "/path/to/your/structure.pdb"
+        stru = sp.get_structure(pdb_filepath)
+
+Execute API
+----------------------------------------------
+
+Use ``preparation.protonate.protonate_stru`` to protonate (i.e. add hydrogen atoms to) your structure.
+
+.. panels::
+
+    :column: col-lg-12 col-md-12 col-sm-12 col-xs-12 p-2 text-left
+
+    The simpliest use of ``protonate_stru`` is as follows.
+        Where the ``ph`` is set to ``7.0``, and ``protonate_ligand`` is set to ``True`` by default.
+
+    .. code:: python
+        
+        from enzy_htp.preparation import protonate
+
+        protonate.protonate_stru(stru=stru)
+    
+    We can also directly import ``protonate_stru`` from ``enzy_htp.preparation`` since it has been cited in
+    the ``__init__.py`` file of ``preparation`` module.
```

Comment:
We dont teach user how python works in our tutorial. Please remove them.

Reply:
Deleted.